proseco/acq.py

```diff
@@ -848,6 +936,49 @@ def calc_p_brightest(self, acq, box_size, man_err=0, bgd=0):
 
         return prob
 
+    def get_overlap_penalties(self):
+        """
+        Get the probability penalties for overlapping boxes.
+
+        This is returned as a multiplicative factor on the acquisition success
+        probability. No overlap penalty implies a value of 1.0.
+
+        :returns: list of penalties (float)
+        """
+        n_acq = len(self)
+        penalties = np.ones(n_acq)
+        if os.environ.get("PROSECO_DISABLE_OVERLAP_PENALTY") == "True":
+            return penalties
+
+        mags = self["mag"].data
+        halfws = self["halfw"].data
+        yangs = self["yang"].data
+        zangs = self["zang"].data
+
+        for idx1 in range(n_acq):
+            mag1 = mags[idx1]
+            halfw1 = halfws[idx1]
+            yang1 = yangs[idx1]
+            zang1 = zangs[idx1]
+
+            for idx2 in range(idx1 + 1, n_acq):
+                mag2 = mags[idx2]
+                halfw2 = halfws[idx2]
+                yang2 = yangs[idx2]
+                zang2 = zangs[idx2]
+                overlap_threshold = halfw1 + halfw2 + OVERLAP_PAD
```

Comment:
I think this is repeating box_overlap().

Reply:
Fixed.